native_client/client.cc

```diff
@@ -238,6 +255,65 @@ ProcessFile(ModelState* context, const char* path, bool show_times)
   }
 }
 
+std::vector<meta_word>
+WordsFromMetadata(Metadata* metadata)
+{
+  std::vector<meta_word> word_list;
+
+  std::string word = "";
+  float word_start_time = 0;
+
+  // Loop through each character
+  for (int i = 0; i < metadata->num_items; i++) {
+    MetadataItem item = metadata->items[i];
+
+    if (strcmp(item.character, " ") != 0) {
+      word.append(item.character);
```

Comment:
As `word` initially contains a non-UTF-8 empty string, see above, how does this append work when `item.character` is UTF-8?

Reply:
`item.character` comes straight from the Alphabet class/text file, if malformed UTF-8 sequences are a worry they should be handled in the Alphabet classes, not here.

Reply:
I think you missed my point or maybe I didn't state it clearly enough.

UTF-8 string literals in C++ can be created by appending a u8 to a normal string literal. So for example en empty UTF-8 string can be created as follows
```C++
std::string word = u8"";
```
A non-UTF-8 empty string can be created as follows
```C++
std::string word = "";
```
which is what is done above.

The question is what happens when we append UTF-8 strings to a non-UTF-8 string
```C++
word.append(item.character);
```
as we are currently doing in this code.

My guess is we get a mess.

Reply:
The empty string literal `""` is valid UTF-8 :)

Reply:
```C++
""
```
is not a UTF-8 string literal
```C++
u8""
```
is a UTF-8 string literal. This is simply a definition, see for example [here](https://en.cppreference.com/w/cpp/language/string_literal).

You are asserting that
```C++
""
```
is equivalent to
```C++
u8""
```
for all compilers.

I don't know if that's the case, but if faced with reading the C++ standard or prefixing "" with u8, I'd prefix "" with u8 and be done with it.

Reply:
Even if "" and u8"" are equivalent u8"" acts a very good documentation of what we expect of word.

Reply:
Fine, point taken, adding u8 is easy enough.